From 585f04f2bf11c84367641f62a5e280a0241a71de Mon Sep 17 00:00:00 2001
From: Derek Schuff <dschuff@chromium.org>
Date: Tue, 7 Jul 2026 12:45:34 -0700
Subject: [PATCH] [Support] Add missing include for MacOS defines (#208065)

https://github.com/llvm/llvm-project/pull/142733 removed several
includes, including
llvm/Support/ProgramStack.h. This was indirectly including headers that
defined
PRIO_DARWIN_THREAD and PRIO_DARWIN_BG.

Add back sys/resource.h to define them locally.
---
 llvm/lib/Support/CrashRecoveryContext.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/llvm/lib/Support/CrashRecoveryContext.cpp b/llvm/lib/Support/CrashRecoveryContext.cpp
index fb9f2ec333a7b..493ba951fd3e1 100644
--- a/llvm/lib/Support/CrashRecoveryContext.cpp
+++ b/llvm/lib/Support/CrashRecoveryContext.cpp
@@ -15,5 +15,8 @@
 #include <cassert>
 #include <mutex>
 #include <setjmp.h>
-
+#ifdef __APPLE__
+#include <sys/resource.h>
+#endif
+
 using namespace llvm;
